wfe2/verify.go

```diff
@@ -395,14 +425,10 @@ func (wfe *WebFrontEndImpl) lookupJWK(
 
 	header := jws.Signatures[0].Header
 	accountURL := header.KeyID
-	prefix := web.RelativeEndpoint(request, acctPath)
-	accountIDStr := strings.TrimPrefix(accountURL, prefix)
```

Comment:
Random question: We don't test that the string actually has the prefix here, does that mean before this change you could just send a bare account ID in the header and it'd just work? It seems unlikely anyone was doing this (probably), and it's not a behavior we should maintain (the new version does the right check), just wondering.

Reply:
>  does that mean before this change you could just send a bare account ID in the header and it'd just work? It seems unlikely anyone was doing this (probably), and it's not a behavior we should maintain (the new version does the right check), just wondering.

Yup - it was an oversight that allowed for more lax parsing of the Key ID than we intended. @rmbolger [noticed this behaviour in Pebble initially](https://github.com/letsencrypt/pebble/issues/109) and also verified that it was the case with Boulder's implementation.

I don't suspect anyone was relying on this behaviour. If they were, they shouldn't have been!